### PR TITLE
litterbox: init at 1.9

### PR DESCRIPTION
**Pull request by adamtrain**
## Description of changes

[Litterbox](https://git.causal.agency/litterbox/about/) is a simple IRC logger that is designed to work with [pounce](https://git.causal.agency/pounce/about/), which is [already in nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/servers/pounce/default.nix#L24).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -601,6 +601,12 @@
     githubId = 1229027;
     name = "Andreas Schrägle";
   };
+  ajwhouse = {
+    email = "hi@adamhouse.net";
+    github = "ajwhouse";
+    githubId = 56616368;
+    name = "Adam House";
+  };
```
Comment (Lurkki14) [inline maintainers/maintainer-list.nix:609]: Adding yourself to maintainers needs to be a separate commit; `maintainers: add ajwhouse`

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -33413,6 +33413,8 @@ with pkgs;
 
   linuxsampler = callPackage ../applications/audio/linuxsampler { };
 
+  litterbox = callPackage ../applications/networking/irc/litterbox { };
+
```
Comment (Lurkki14) [inline pkgs/top-level/all-packages.nix:33417]: New packages are recommended to go to `pkgs/by-name` https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

```suggestion

```

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (ajwhouse) [reply pkgs/top-level/all-packages.nix]: Ah, sorry about that. I got caught up in doing it a way symmetric with `pounce`. I have corrected this and re-tested to satisfaction.

File: pkgs/by-name/li/litterbox/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  buildInputs = [ libressl sqlite ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildFlags = [ "all" ];
+
+  meta = with lib; {
+    homepage = "https://code.causal.agency/june/litterbox";
+    description = "Simple TLS-only IRC logger";
+    license = licenses.gpl3;
```
Comment (Lurkki14) [inline pkgs/by-name/li/litterbox/package.nix:21]:
```suggestion
    license = licenses.gpl3Plus;
```
Just `licenses.gpl3` is deprecated

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Review by Lurkki14 — APPROVED** [review]

**Review by sinanmohd — APPROVED** [review]

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by ajwhouse** [comment]
@AndersonTorres appreciate your detailed feedback; made all changes you requested; excited to submit more packages and I will attend to all these details proactively in the future.

**Review by AndersonTorres — APPROVED** [review]
